### PR TITLE
Compare Webpack to SystemJS

### LINKED ISSUE #152: Comparison - Talk About webpack vs SystemJS
_From @TheLarkInn on July 1, 2016 12:47_

**I'm submitting a feature request**

**Webpack version:**
1.10.x/2.x

There has been a lot of community confusion in regards to SystemJS and Webpack. This issue is a stub to add some documentation talking about the difference between a module bundler and a module loader and how they perform 2 different things.

**Please tell us about your environment:**
OSX 10.x / Linux / Windows 10

**Current behavior:**

**Expected/desired behavior:**
- **What is the expected behavior?**
- **What is the motivation / use case for changing the behavior?**
- **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ] 
- **Language:** [all | TypeScript X.X | ES6/7 | ES5 | Dart | ...] 

_Copied from original issue: webpack/webpack#2727_

### PR DESCRIPTION
**Pull request by skipjack**
Add a few related articles and a section about the difference between loading and transpiling at runtime vs processing and bundling beforehand. I'd like to do some other cleanup in this article in terms of the table (and potentially move the whole thing out of `/guides`) but I stashed those changes for now as they'll require a bit more thought.

Resolves #152

### CODE REVIEW

File: content/guides/comparison.md [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
+title: Comparison
 sort: 25
 contributors:
   - pksjce
   - bebraw
   - chrisVillanueva
   - tashian
   - simon04
+related:
+  - title: JSPM vs. Webpack
```
Comment (simon04) [inline content/guides/comparison.md]: We mostly also use lowercase "webpack" for external page titles, e.g. see d191256b0ff58a51104f47cf377b51a90ce518a4.

File: content/guides/comparison.md [hunk 2]
```diff
@@ -46,3 +51,10 @@ webpack is not the only module bundler out there. If you are choosing between us
 ♦ in production mode (opposite in development mode)
 
 X is the length of the path string
+
+
+## Bundling vs. Loading
+
+It's important to note some key differences between _loading_ and _bundling_ modules. A tool like [SystemJS](https://github.com/systemjs/systemjs), which can be found under the hood of [JSPM](http://jspm.io/), is used to load and transpile modules at runtime in the browser. This differs significantly from webpack, where modules are transpiled (through "loaders") and bundled before hitting the browser.
+
+Each method has it's pros and cons. Loading and transpiling modules at runtime can add a lot of overhead for larger sites and applications comprised of many modules. For this reason, SystemJS makes more sense for smaller projects where fewer modules are required. However, this may change as load times decrease and [HTTP/2](https://http2.github.io/) gains support.
```
Comment (simon04) [inline content/guides/comparison.md]: Maybe you can differentiate between transfer from server and compiling/processing on client more clearly. HTTP/2 only improves the former. The latter is especially important for mobile devices.

**Comment by skipjack** [comment]
@simon04 thanks for the review, I updated the warning. Please skim again when you have a chance.

File: content/guides/comparison.md [hunk 3]
```diff
@@ -57,4 +57,4 @@ X is the length of the path string
...
 
-Each method has it's pros and cons. Loading and transpiling modules at runtime can add a lot of overhead for larger sites and applications comprised of many modules. For this reason, SystemJS makes more sense for smaller projects where fewer modules are required. However, this may change as load times decrease and [HTTP/2](https://http2.github.io/) gains support.
+Each method has it's pros and cons. Loading and transpiling modules at runtime can add a lot of overhead for larger sites and applications comprised of many modules. For this reason, SystemJS makes more sense for smaller projects where fewer modules are required. However, this may change a bit as [HTTP/2](https://http2.github.io/) will improve the speed at which files can be transferred from server to client gains support. Note that HTTP/2 doesn't change anything about _transpiling_ modules, which will always take longer when done client-side.
```
Comment (simon04) [inline content/guides/comparison.md]: "gains support" is superfluous.

Comment (skipjack) [inline content/guides/comparison.md]: Done.. good catch thanks.

**Pull request merged**